[AngJunYang]iP

AngJunYang opened the pull request:
**Duke Pro**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

AngJunYang committed:
Level-1

AngJunYang committed:
Level 2. Add, List

Added the ability to store whatever text entered by the user and display them back to the user when requested.

AngJunYang committed:
Level 3. Mark as Done

Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

AngJunYang committed:
Level 4. ToDos, Events, Deadlines

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019
Example:

AngJunYang committed:
Automated UI Testing

AngJunYang committed:
Handle Errors

AngJunYang committed:
Delete

AngJunYang committed:
Level-7

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when Duke starts up. You may hard-code the file name and location

AngJunYang committed:
Level-8

Teach Duke how to understand dates and times. For example, if the command is deadline return book /by 2/12/2019 1800, Duke should understand 2/12/2019 1800 as 2nd of December 2019, 6pm, instead of treating it as just a String.

AngJunYang committed:
Merge branch 'branch-Level-8'

AngJunYang committed:
A-moreOOP

AngJunYang committed:
A-Packages

AngJunYang committed:
Merge branch 'add-gradle-support'

AngJunYang committed:
A-JUnit

AngJunYang committed:
A-Jar

AngJunYang committed:
A-JavaDoc

AngJunYang committed:
A-CodingStandard

AngJunYang committed:
Level-9

AngJunYang committed:
Merge branch 'branch-A-JavaDoc'

AngJunYang committed:
Merge branch 'branch-Level-9'

chia-yh commented on src/main/java/Duke/Chatbot/Messages.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public class Messages {
+
+    final static String MESSAGE_END = "GOODBYE";
```
I like that the associated constants have a common prefix.

chia-yh commented on src/main/java/Duke/Chatbot/Storage.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.List;
+import java.util.Scanner;
+
+public class Storage {
```
Perhaps include some descriptive header comments for this class?

chia-yh commented on src/main/java/Duke/Tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,57 @@
+ */
+public class DeadlineTask extends Task {
+
+    LocalDateTime dueDate;
```
Perhaps add some JavaDoc for class members?

chia-yh commented on src/main/java/Duke/Chatbot/Ui.java:
```diff
@@ -0,0 +1,217 @@
+/**
+ * Handles the output to be printed to user.
+ */
+public class Ui {
```
I like that the acronym used as the class name is not uppercase.

chia-yh commented on src/main/java/Duke/Chatbot/Parser.java:
```diff
@@ -0,0 +1,166 @@
+ * Converts user input to make sense of user command
+ */
+public class Parser {
+    public static final Map<String, BiPredicate<Chatbot, String>> commands
```
I like the use of line break to improve readability.

chia-yh reviewed (commented):
Looks good, pretty much no coding standard issues, keep up the good work!

jweeyh commented on src/main/java/Duke/Chatbot/Parser.java:
```diff
@@ -0,0 +1,166 @@
+     * @param nextLine Next line of User Input
+     */
+    public void parse(String nextLine) {
+        //Assuming commands start with a space.
```
I like the use of comments to make your code clearer to readers

jweeyh commented on src/main/java/Duke/Chatbot/Ui.java:
```diff
@@ -0,0 +1,217 @@
+    /**
+     * Message to be shown to user on startup
+     */
+    public void showStartupMessage() {
```
I like the creative logo design :)

jweeyh commented on src/main/java/Duke/Chatbot/Ui.java:
```diff
@@ -0,0 +1,217 @@
+     * is currently unmarked
+     */
+    public void showTaskStateIncompletedError() {
+        System.out.println("Task is already marked as incompelted.");
```
Minor spelling error

jweeyh commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,60 @@
+Deletion complete!
+There are no tasks to print!
+Deletion complete!
+There are no tasks to print!
```
I like the inclusion of expected text.

jweeyh reviewed (commented):
Excellent effort put into documentation, testing and abstraction of classes. Keep it up!

AngJunYang committed:
A-CheckStyle

nigelteosw commented on src/main/java/Duke/Chatbot/Parser.java:
```diff
@@ -130,6 +130,14 @@ private void setupCommands() {
             return true;
         });
 
+        commands.put(Messages.MESSAGE_FIND, (chatbot, args) -> {
```
Clean way of implementing the find function, good job!

nigelteosw commented on src/main/java/Duke/Chatbot/Ui.java:
```diff
@@ -46,68 +70,128 @@ public void showStartupMessage() {
+     * Error message to be shown when task the user is trying to mark
+     * has already been marked
+     */
     public void showTaskStateCompletedError() {
```
I find it nice that you added more specific errors to be more clear to the user.

nigelteosw reviewed (commented):
Looks good to me, keep it up

AngJunYang committed:
Level-10

AngJunYang committed:
Add Assertions

Add Assert
Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

AngJunYang committed:
Merge pull request #2 from AngJunYang/branch-Assertions

Add Assertions

AngJunYang committed:
Added Ui.png

AngJunYang committed:
link GUI to commands

AngJunYang committed:
Remove very long method in Parser

AngJunYang committed:
Update ReadMe

AngJunYang committed:
Update ReadMe to reflect v0.2 instructions
